ffmpeg_7: 7.0 -> 7.0.1

jopejoe1 opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jopejoe1 committed:
ffmpeg_7: 7.0 -> 7.0.1

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -464,14 +464,6 @@ stdenv.mkDerivation (finalAttrs: {
         url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/3bb00c0a420c3ce83c6fafee30270d69622ccad7";
         hash = "sha256-oZMZysBA+/gwaGEM1yvI+8wCadXWE7qLRL6Emap3b8Q=";
       }
-    ])
-    ++ (lib.optionals (lib.versionAtLeast version "7.0") [
-      {
-        # Will likely be obsolete in >7.0
-        name = "fate_avoid_dependency_on_samples";
-        url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/7b7b7819bd21cc92ac07f6696b0e7f26fa8f9834";
-        hash = "sha256-TKI289XqtG86Sj9s7mVYvmkjAuRXeK+2cYYEDkg6u6I=";
-      }
```
This does not appear to be fixed upstream yet somehow?

Atemu replied:
Result of `nixpkgs-review pr 316998` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>handbrake</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>ffmpeg_7</li>
    <li>ffmpeg_7-full</li>
    <li>ffmpeg_7-full.bin</li>
    <li>ffmpeg_7-full.data</li>
    <li>ffmpeg_7-full.dev</li>
    <li>ffmpeg_7-full.doc</li>
    <li>ffmpeg_7-full.lib</li>
    <li>ffmpeg_7-full.man</li>
    <li>ffmpeg_7-headless</li>
    <li>ffmpeg_7-headless.bin</li>
    <li>ffmpeg_7-headless.data</li>
    <li>ffmpeg_7-headless.dev</li>
    <li>ffmpeg_7-headless.doc</li>
    <li>ffmpeg_7-headless.lib</li>
    <li>ffmpeg_7-headless.man</li>
    <li>ffmpeg_7.bin</li>
    <li>ffmpeg_7.data</li>
    <li>ffmpeg_7.dev</li>
    <li>ffmpeg_7.doc</li>
    <li>ffmpeg_7.lib</li>
    <li>ffmpeg_7.man</li>
  </ul>
</details>

Atemu replied:
Ohhh, it's the handbrake ffmpeg. Confusingly the same name.

It's still on ABI 7.0, so let's simply keep the patch for now.

jopejoe1 replied:
Re-added the patch for version 7.0, fails to apply on 7.0.1

jopejoe1 commented:
x86_64-darwin build is still broken even after the update.

Atemu reviewed (changes requested):
This still breaks handbrake's ffmpeg because they use overrideAttrs to pass the version which ffmpeg does not use anymore.

I'll create a patch for that.

Atemu committed:
handbrake: use ffmpeg's new override interface

ffmpeg no longer responds to changing version using overrideAttrs. This causes
an issue where a critical patch that is included when ABI == 7.0 would not be
applied because the ffmpeg drv still thinks it's building for ABI 7.0.1.

You must specify the ABI version via the explicit override API now which this
patch does.

The ffmpeg derivation now supports fetching an upstream source for a certain
version; there is no need to do so manually anymore.

Atemu reviewed (approved):
Handbrake now also works.
